src/kzg.rs

```diff
@@ -295,6 +293,8 @@ impl Kzg {
         let mut buffer = vec![0u8; point_size];
 
         let mut i = 0;
+        // We are making one syscall per field element, which is super inefficient.
+        // FIXME: read the entire file into memory and then split it into field elements.
```

Comment:
The entire G1 point file is 8GB or so i believe. We might need a middle ground.

Reply:
ah good point ok amended comment: [bef408a](https://github.com/Layr-Labs/rust-kzg-bn254/pull/16/commits/bef408a74da3167cd9ab0c447eab2d547b8b21f9)

Reply:
Wait thinking about this more... if the file is 8GB isnt it also going to take >= 8GB in RAM? Or is there some form of compression we're doing after loading?